src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -989,6 +999,13 @@ private TypedConstant VisitConversion(BoundConversion node, BindingDiagnosticBag
                 var operand = node.Operand;
                 var operandType = operand.Type;
 
+                if (node.Conversion.IsCollectionExpression
+                    && node.Conversion.GetCollectionExpressionTypeKind(out _) == CollectionExpressionTypeKind.Array)
```

Comment:
This seems likely to cause a binding cycle. `GetCollectionExpressionTypeKind` calls `GetEarlyDecodedWellKnownAttributeData`, which forces binding of early attributes, which we're already in the middle of doing here. `GetCollectionExpressionTypeKind` should likely be updated to have an `isEarlyBinding` flag and avoid doing anything but checking for array in such cases.

Reply:
This is `Conversion.GetCollectionExpressionTypeKind` which doesn't pull on `GetEarlyDecodedWellKnownAttributeData`.